pallets/communities/src/lib.rs

```diff
@@ -13,96 +179,202 @@ mod tests;
-	// These functions materialize as "extrinsics", which are often compared to transactions.
-	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	// Dispatchable functions allows users to interact with the pallet and invoke
+	// state changes. These functions materialize as "extrinsics", which are often
+	// compared to transactions. Dispatchable functions must be annotated with a
+	// weight and must return a DispatchResult.
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		/// An example dispatchable that takes a singles value as a parameter, writes the value to
-		/// storage and emits an event. This function must be dispatched by a signed extrinsic.
+		/// Registers an appliation as a new community, taking an
+		/// [existential deposit][9] used to create the community account.
+		///
+		/// [9]: https://docs.substrate.io/reference/glossary/#existential-deposit
 		#[pallet::call_index(0)]
-		#[pallet::weight(T::WeightInfo::do_something())]
-		pub fn do_something(origin: OriginFor<T>, something: u32) -> DispatchResult {
-			// Check that the extrinsic was signed and get the signer.
-			// This function will return an error if the extrinsic is not signed.
-			// https://docs.substrate.io/main-docs/build/origins/
+		#[pallet::weight(T::WeightInfo::apply())]
+		pub fn apply(origin: OriginFor<T>, community_id: T::CommunityId) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			// Update storage.
-			<Something<T>>::put(something);
+			Self::do_register_community(&who, &community_id)?;
+			Self::do_create_community_account(&who, &community_id)?;
 
 			// Emit an event.
-			Self::deposit_event(Event::SomethingStored { something, who });
+			Self::deposit_event(Event::CommunityCreated { id: community_id, who });
+
 			// Return a successful DispatchResultWithPostInfo
 			Ok(())
 		}
 
-		/// An example dispatchable that may throw a custom error.
+		/// Sets some [`CommunityMetadata`][11] to describe the
+		/// community.
+		///
+		/// [11]: `types::CommunityMetadata`
 		#[pallet::call_index(1)]
-		#[pallet::weight(T::WeightInfo::cause_error())]
-		pub fn cause_error(origin: OriginFor<T>) -> DispatchResult {
-			let _who = ensure_signed(origin)?;
-
-			// Read a value from storage.
-			match <Something<T>>::get() {
-				// Return an error if the value has not been set.
-				None => return Err(Error::<T>::NoneValue.into()),
-				Some(old) => {
-					// Increment the value read from storage; will error in the event of overflow.
-					let new = old.checked_add(1).ok_or(Error::<T>::StorageOverflow)?;
-					// Update the value in storage with the incremented result.
-					<Something<T>>::put(new);
-					Ok(())
+		#[pallet::weight(T::WeightInfo::set_metadata())]
+		pub fn set_metadata(
+			origin: OriginFor<T>,
+			community_id: T::CommunityId,
+			name: Option<Field<64>>,
+			description: Option<Field<256>>,
+			urls: Option<BoundedVec<Field<2000>, T::MaxUrls>>,
```

Comment:
I would create separate constants (Get<u32>) for each and let the definition during runtime configuration.

Reply:
Yup. Done.